spark/dl/src/main/scala/com/intel/analytics/bigdl/tensor/DenseTensorMath.scala

```diff
@@ -394,6 +394,18 @@ object DenseTensorMath {
     self
   }
 
+  def prodAll[@specialized(Float, Double) T](self: DenseTensor[T])(
+    implicit ev: TensorNumeric[T]): T = {
+    var product = ev.fromType[Int](1)
+    val func = new TensorFunc2[T] {
```

Comment:
`prodAll()` is not used in any layer, just keep consistent with sumAll and similar implementation.
For the `Prod` operations, it will call `prod` method, which using `dimApply2`.